src/braket/pennylane_plugin/translation.py

```diff
@@ -698,6 +707,14 @@ def translate_result(
             ag_result.value["gradient"][f"p_{i}"]
             for i in sorted(key_indices)
         ]
+
+    if measurement.return_type is ObservableReturnTypes.Sample and observable is None:
+        if isinstance(measurement, qml.measurements.SampleMeasurement):
+            if targets:
+                return [m[targets] for m in braket_result.measurements]
+            return braket_result.measurements
+        raise NotImplementedError(f"Unsupported measurement type: {type(measurement)}")
```

Comment:
same question here, would it be possible to embed this logic inside `translate_result_type` somehow?